### DISCUSSION TITLE
To find r2score of my model

### DISCUSSION THREAD

**Discussion by bibinwils** [Lightning-AI/pytorch-lightning#6125] [code help: CV] [answered]
I have a UNet model. I'm trying for a regression model since, in my output, I have different floating values for each pixel. In order to check the r2score, I tried to put the below code in the 'model class', training_step, validation_step, and test_step. 

`from pytorch_lightning.metrics.functional import r2score`
`r2 = r2score(logits, y)`
`self.log('r2:',r2)`

But it's giving the following error

> ValueError: Expected both prediction and target to be 1D or 2D tensors, but recevied tensors with dimension torch.Size([50, 1, 32, 32])

How can I check my model fit?

**Comment by SkafteNicki** [comment, marked as answer]
I assume that you just want to calculate the total score, and in that case you should simply flatten your input before calculating the score:
``` python
from pytorch_lightning.metrics.functional import r2score
r2 = r2score(logits.flatten(), y.flatten())
self.log('r2:',r2)
```
